### PR TITLE
[Auto] Update interface for sparsifyml

### PR DESCRIPTION
**Pull request by KSGulin**
This PR updates the sparsify.auto interface to be compatible with a local run via sparsifyml.

Changes made
- Classes used by both `sparsify.auto` and the backend have been moved into the new `sparsify.schemas` to avoid import cycles
- Renamed `sparsify.auto.api` to `sparsify.auto.scripts`, as it only contains the main script now. Note that there is an integration-related namespace clash that prevents us from being able to move main outside of an isolated directory
- Explicit torch version requirement removed, as we already have sparseml-tied torch requirement with `sparseml[torchvision]`
- Updated base version to `1.4.0`

Test plan
`Make test TARGETS=auto`

**NOTE:** We're running into a yolov5 related issue on one of the tests, as the integration has changed since the last time sparsify.auto has been updated. There will be a separate effort to fix this issue which will also involved fixing currently failing integration tests

### CODE REVIEW

**Review by bfineran — DISMISSED** [review]
approved, any reason for `interface` for the module name instead of `schemas` or something similar? thinking just because interface has oop connotations

**Comment by KSGulin** [comment]
@bfineran Good call! Updated

**Review by bfineran — APPROVED** [review]

File: src/sparsify/auto/tasks/runner.py [hunk 1]
```diff
@@ -79,7 +78,6 @@ def _wrapper(self, *args, **kwargs):
 
             # attempt run and catch errors until success or maximum number of attempts
             # exceeded
-            return func(self, *args, **kwargs)
             while not error_handler.max_attempts_exceeded():
```
Comment (rahul-tuli) [inline src/sparsify/auto/tasks/runner.py:81]: Why was this line removed?

File: utils/docs_builder.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def _copy_docs_data(src: str, dest: str, folders: List[str]):
 
     # loop through version folders to add data to, None represents root folder
     for ver_folder in [None, *folders]:
-        for (map_from, map_to) in map_dirs:
```
Comment (rahul-tuli) [inline utils/docs_builder.py:105]: nice

**Review by rahul-tuli — APPROVED** [review]
LGTM!

File: src/sparsify/auto/tasks/runner.py (hunk 1 above)
Comment (KSGulin) [reply src/sparsify/auto/tasks/runner.py:81]: It was a debug line that shouldn't have made it into the branch
